patches/tModLoader/Terraria/GameContent/Profiles.TML.cs

```diff
@@ -0,0 +1,47 @@
+using Microsoft.Xna.Framework.Graphics;
+using ReLogic.Content;
+using Terraria.ModLoader;
+
+namespace Terraria.GameContent;
+
+public partial class Profiles
+{
+	/// <summary>
+	/// Class that is some-what identical to <seealso cref="LegacyNPCProfile"/> that allows for
+	/// modded texture usage. Also allows for any potential children classes to mess with the fields.
+	/// </summary>
+	public class DefaultNPCProfile : ITownNPCProfile
+	{
+		protected string rootFilePath;
```

Comment:
No benefit to storing this, since it's not used outside the constructor. Subclasses can obtain `npcTexturePath` from the constructor, or from `defaultTexture.Path`